src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -2468,6 +2470,49 @@ internal sealed override bool HasInlineArrayAttribute(out int length)
             return false;
         }
 
+#nullable enable
+        internal sealed override bool HasCollectionBuilderAttribute(out TypeSymbol? builderType, out string? methodName)
+        {
+            var uncommon = GetUncommonProperties();
+            if (uncommon == s_noUncommonProperties)
+            {
+                builderType = null;
+                methodName = null;
+                return false;
+            }
+
+            if (ReferenceEquals(uncommon.lazyCollectionBuilderAttributeData, CollectionBuilderAttributeData.Uninitialized))
```

Comment:
Nit: you used an `(object)` cast above, and `ReferenceEquals` here. I have a slight personal preference towards `ReferenceEquals`, but please use one 😉.